.travis.yml

```diff
@@ -3,7 +3,7 @@ language: node_js
 node_js:
   # we recommend testing addons with the same minimum supported node version as Ember CLI
   # so that your addon works for all apps
-  - '6'
+  - "8"
```

Comment:
Bumping the engine actually isn't necessary.

I just verified that we can make these changes without updating `ember-cli`/`ember-source`/etc (including the blueprint update which is what causes the engine update) if we use `ember-concurrency@1.0.0` and `ember-concurrency-decorators@0.6.0` in the tests (since anything newer won't run on Node 6). I'd prefer to run against the latest semver-compatibly 1.0s of these, but given that the code doesn't actually integrate with `e-c` directly, it's just the most common expected usage of `waitForCoroutine`, this is a mild preference.

As far as the major version bump goes, I guess I'd challenge the assumption that we'd need to do another major version bump in the next month. I'm following `ember-cli`'s blueprints here, and `ember-cli` dropped Node 6 from the blueprint 7 months ago, but we haven't had to update this library's blueprint/Node support until now (and per my investigation above, we don't even *have* to do it now). Are you anticipating some specific feature of Node 10 or 12 that we'll want to use in the next month that will require dropping support for Node 8? If not, then isn't it reasonable to assume that we could wait until at least April when Node 10 end-of-lifes before having to do another major version bump?

You've helped me realize that the Ember update is a separate change from this one, although it allows a cleaner/safer testing story IMO. It's just not clear to me that we gain anything by waiting for drop support for Node 6.

But I'm game either way:
1. Break out the `ember-cli`/blueprint update into a separate PR that gets merged before this one
2. Drop the `ember-cli`/blueprint update and use older versions of the `e-c` libraries for testing.

Reply:
Ya so typically we try to move node versions in lockstep with current LTS versions. As of 12/31, node 8 will be EOL (https://nodejs.org/en/about/releases/). 

If this change requires dropping node 6, and I do believe it does as the floating dependency tests will fail without that change due to another library removing node 6 support but not major version bumping, we're going to have to do a major version bump in _this_ library. Once node 8 EOLs, we're going to have to do another major bump. 

Does that make sense?

Reply:
Okay. I'm not following why another major bump will be required when Node 8 EOLs given that Node 6 EOLed 7 months ago and we haven't major bumped yet...but no worries, I don't need to understand and definitely don't want to be [this guy](https://xkcd.com/386/)!

I'm 90% sure that this change doesn't require dropping Node 6. The floating dependencies builds have been passing up until now (in a shocking and unprecedented plot twist, semver appears to be working!), and I can change the only dependencies that I need to add (`e-c` and `e-c-decorators` as `devDependencies`) to be pinned to versions that support Node 6. So we can stay on Node 6 until Node 8 EOLs, `ember-cli` updates the blueprint to drop 8 from `engines`, and then we can `ember-cli-update` at that point and just skip right over Node 8.

I'll make a backup copy of this branch, and then rebase away the `ember-cli`/blueprint update, pin the versions, and turn this into a more narrow PR.

Reply:
(BTW, I just ran `ember try:one ember-default` locally with the dependency pinning I mentioned, and it passed, so now 95% sure we can merge this and stay on Node 6).

Reply:
We discussed this offline, and can fix this by simply updating the travis.yml to run against node 8. This is done in another PR, which should likely get merged first.